main/src/ca/uwaterloo/flix/language/phase/Resolver.scala

```diff
@@ -1481,4 +1495,14 @@ object Resolver extends Phase[NamedAst.Root, ResolvedAst.Root] {
     Name.NName(sp1, idents, sp2)
   }
 
+  /**
+    * Tries to apply `tpe1` to `tpe2`. Creates a resolution error if `tpe1` is not a type constructor.
+    */
+  private def tryMkApply(tpe1: Type, tpe2: Type, loc: SourceLocation): Validation[Type, ResolutionError] = {
```

Comment:
Actually, let us just call this `mkApply`. We usually dont use `tryMk` even when the operation can fail. Thats implicit in returning a validation.